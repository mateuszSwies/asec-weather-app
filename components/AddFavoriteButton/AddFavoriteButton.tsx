import { styles } from './styles';
import { FAB } from 'react-native-paper';

type AddFavoriteButtonProps = {
	onFavoriteButtonPress: () => void;
};

const AddFavoriteButton = ({
	onFavoriteButtonPress,
}: AddFavoriteButtonProps) => {
	return (
		<FAB
			icon="heart"
			style={styles.fab}
			label="Favorites"
			onPress={onFavoriteButtonPress}
		/>
	);
};

export default AddFavoriteButton;
