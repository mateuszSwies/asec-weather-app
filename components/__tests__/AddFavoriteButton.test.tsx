import React from 'react';
import { render, fireEvent, act } from '@testing-library/react-native';
import AddFavoriteButton from '../AddFavoriteButton/AddFavoriteButton';

describe('AddFavoriteButton', () => {
	test('renders correctly', () => {
		const { getByLabelText } = render(
			<AddFavoriteButton onFavoriteButtonPress={() => {}} />
		);
		const button = getByLabelText('Favorites');
		expect(button).toBeDefined();
	});

	test('calls onFavoriteButtonPress when button is pressed', () => {
		const onFavoriteButtonPress = jest.fn();
		const { getByLabelText } = render(
			<AddFavoriteButton onFavoriteButtonPress={onFavoriteButtonPress} />
		);
		const button = getByLabelText('Favorites');
		act(() => {
			fireEvent.press(button);
		});
		expect(onFavoriteButtonPress).toHaveBeenCalled();
	});
});
