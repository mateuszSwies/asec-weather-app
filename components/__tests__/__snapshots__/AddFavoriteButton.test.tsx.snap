// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFavoriteButton renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(234, 221, 255, 1)",
      "borderRadius": 16,
      "bottom": 0,
      "margin": 16,
      "opacity": 1,
      "position": "absolute",
      "right": 0,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="fab-container-outer-layer"
>
  <View
    collapsable={false}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "rgba(234, 221, 255, 1)",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="fab-container"
  >
    <View
      accessibilityLabel="Favorites"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 16,
          },
        ]
      }
      testID="fab"
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            {
              "borderRadius": 16,
              "height": 56,
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="fab-content"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "justifyContent": "center",
              },
              {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": [
                  {
                    "rotate": "0deg",
                  },
                ],
              }
            }
            testID="cross-fade-icon-current"
          >
            <Text />
          </View>
        </View>
        <Text
          selectable={false}
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                [
                  {
                    "marginHorizontal": 8,
                  },
                  false,
                  {
                    "color": "rgba(33, 0, 93, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                ],
              ],
            ]
          }
          testID="fab-text"
        >
          Favorites
        </Text>
      </View>
    </View>
  </View>
</View>
`;
